k8s/upgrade/src/bin/upgrade-job/upgrade/utils.rs

```diff
@@ -7,12 +7,42 @@ use crate::common::{
 };
 use k8s_openapi::api::core::v1::Pod;
 use kube::{api::ObjectList, ResourceExt};
+use openapi::models::{Volume, VolumeStatus};
 use semver::{Version, VersionReq};
 use snafu::ResultExt;
+use std::time::Duration;
 use tracing::{info, warn};
 
 /// Function to check for any volume rebuild in progress across the cluster
 pub(crate) async fn is_rebuilding(rest_client: &RestClientSet) -> Result<bool> {
+    loop {
+        let unhealthy_volumes = list_unhealthy_volumes(rest_client).await?;
```

Comment:
This will insert unhealthy volume again and again even after timed wait. Please rather maintain a map of timed wait completed volumes and exclude them gradually.

Reply:
done